### PR TITLE
Update the ProfileView & add documentation

### PR DESCRIPTION
**Pull request by alexander-san**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/favors/utils/email/RetrofitClient.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 import static java.net.HttpURLConnection.HTTP_OK;
 
-class RetrofitClient extends RetrofitDispatcher{
+public class RetrofitClient extends RetrofitDispatcher{
```
Comment (alexander-san) [inline app/src/main/java/ch/epfl/sweng/favors/utils/email/RetrofitClient.java]: sorry about this, probably we should consider using reflection
But kept getting access errors so went with this for now (are there any specific concerns with this Jeremey?)

Comment (Jmion) [reply app/src/main/java/ch/epfl/sweng/favors/utils/email/RetrofitClient.java]: Well there is some issues with calling it properly. It is much better to have it in package private. What are the issues in particular?

Comment (AndreaScalisi) [reply app/src/main/java/ch/epfl/sweng/favors/utils/email/RetrofitClient.java]: Where exactly did you get access errors with RetrofitClient ?

File: app/src/androidTest/java/ch/epfl/sweng/favors/utils/UtilsTest.java [hunk 2]
```diff
@@ -219,4 +224,17 @@ public void CreateUtils(){
         new Utils();
     }
 
+    @Test
```
Comment (Jmion) [inline app/src/androidTest/java/ch/epfl/sweng/favors/utils/UtilsTest.java]: Not the right  class for this. Create a package with this name `app/src/androidTest/java/ch/epfl/sweng/favors/utils/app/src/androidTest/java/ch/epfl/sweng/favors/utils/email`
You will not have any issues with accessing package private stuff from there. So you can revert the change of access of the retrofit client back to how it was.

File: app/src/main/java/ch/epfl/sweng/favors/utils/email/RetrofitClient.java (hunk 1 above)
Comment (Jmion) [reply app/src/main/java/ch/epfl/sweng/favors/utils/email/RetrofitClient.java]: ## This is not OK. Read comment bellow about why
